modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/field-sets/FieldSetModal.js

```diff
@@ -0,0 +1,472 @@
+import {DndProvider} from 'react-dnd';
+import {HTML5Backend} from 'react-dnd-html5-backend';
+
+import getFieldsWithoutOptions from '../../../js/components/field-sets/actions/getFieldsWithoutOptions.es';
+import {usePropagateFieldSet} from '../../../js/components/field-sets/actions/usePropagateFieldSet.es';
+import TranslationManager from '../../../js/components/translation-manager/TranslationManager.es';
+import {addItem, updateItem} from '../../../js/utils/client.es';
+import {getFieldSetDDMForm} from '../../../js/utils/dataConverter.es';
+import {isDataLayoutEmpty} from '../../../js/utils/dataLayoutVisitor.es';
+import {normalizeDataLayout} from '../../../js/utils/normalizers.es';
+import {errorToast, successToast} from '../../../js/utils/toast.es';
+import {FormBuilder} from '../../FormBuilder';
+import {EVENT_TYPES} from '../../eventTypes';
+import sidebarReducer from '../../reducers/sidebarReducer';
+
+const DEFAULT_DATA_DEFINITION = {
+	dataDefinitionFields: [],
+};
+
+const DEFAULT_DATA_LAYOUT = {
+	dataLayoutFields: {},
+	dataLayoutPages: [],
+	dataRules: [],
+	paginationMode: 'single-page',
+};
+
+const ModalContent = ({
+	fieldSet: fieldSetProp,
+	onClose,
+	onUpdate,
+	updateFieldSetList,
+}) => {
+	const dispatch = useForm();
+	const {contentType} = useConfig();
+	const {
+		dataDefinitionId,
+		defaultLanguageId,
+		editingLanguageId,
+	} = useFormState();
+	const {dataDefinition, dataLayout} = useFormState({
+		schema: ['dataDefinition', 'dataLayout'],
+	});
+
+	const [name, setName] = useState(dataDefinition.name ?? {});
+	const changeEditingLanguageId = (editingLanguageId) =>
+		dispatch({
+			payload: {editingLanguageId},
+			type: EVENT_TYPES_CORE.LANGUAGE.CHANGE,
+		});
+
+	const onSave = async () => {
+		const {
+			availableLanguageIds,
+			dataDefinitionFields,
+		} = dataDefinition.build();
+
+		const {dataLayoutPages} = dataLayout.build();
+
+		try {
+			if (!name[defaultLanguageId]) {
+				changeEditingLanguageId(defaultLanguageId);
+				throw new Error(
+					Liferay.Language.get('please-enter-a-valid-title')
+				);
+			}
+
+			const fieldsWithoutOptions = getFieldsWithoutOptions(
+				dataDefinitionFields,
+				defaultLanguageId
+			);
+
+			if (fieldsWithoutOptions.length) {
+				throw new Error(
+					Liferay.Util.sub(
+						Liferay.Language.get(
+							'at-least-one-option-should-be-set-for-field-x'
+						),
+						fieldsWithoutOptions[0].label[defaultLanguageId]
+					)
+				);
+			}
+
+			if (dataDefinitionId) {
+
+				// update fieldSet
+
+				const updatedFieldSet = {
+					...fieldSetProp,
+					availableLanguageIds,
+					dataDefinitionFields,
+					defaultDataLayout: normalizeDataLayout(
+						{
+							...fieldSetProp.defaultDataLayout,
+
+							dataLayoutPages: dataLayoutPages.map((page) => {
+								return {
+									...page,
+									description: {
+										[defaultLanguageId]: '',
+									},
+
+									title: {
+										[defaultLanguageId]: '',
+									},
```

Comment:
If this needs to be a default value, treat it within the Schema because it is the data structure contract. Schema tries to take from `localizedDescription` and `localizedTitle`, if it doesn’t exist try to add that there.

Remember that Schema is our contract that transforms the old structure to the new structure, value behaviors, and data structure must be added there to avoid doing things like that and mainly to be manually converting the data. Always try to avoid making these transformations within the components, the easier it will be for you to be able to remove the old structure.